doc/codebase_anatomy.md

```diff
@@ -0,0 +1,54 @@
+# Codebase Anatomy
+
+This document describes the the file structure of the `kb_sdk` codebase.
+
+#### Root level
+
+* `build.xml` - main Ant build configuration file for compiling, running tests, etc.
+* `bin/` - holds the executable files generated on compilation (most importantly, `kb-sdk`)
+* `doc/` - additional documentation about this codebase
+* `Dockerfile` - the docker configuration for the container that runs the SDK
+* `entrypoint` - the entrypoint bash script that is run for the SDK docker container
+* `JAR_DEPS` and `JAR_DEPS_BIN` - indexes of java jar dependencies
+* `javacc/` - shell scripts used to generate code related to parsing KIDL specs
+* `KBaseJobService.spec` - KIDL spec for a JSON RPC job service
+* `lib` - some generic libraries that get included in SDK apps
+* `Makefile` - commands for compiling, building docker images, and initializing submodules
+* `Pipfile` and `Pipfile.lock` - python dependencies for pipenv
+* `reports/` - files generated for JaCoCo test coverage reports
+* `sdkbase/` - Docker files for the image used inside actual SDK apps
+* `src/` - the main source code for this project; see below
+* `submodules/` and `submodules_hacks` - jars and other dependencies
+* `test_scripts/` - test helpers in perl, python, and js
+
+#### Source code in `/src/java/us/kbase`
+
+* `catalog/` - Services for managing, registering, and building modules in the catalog. All code here is duplicated in the `njs_wrapper`
+* `common/executionengine/` - Code for executing jobs and sub-jobs. All the code here is duplicated in the `njs_wrapper` repo
+* `common/service/` - Some tuple datatypes
+* `common/utils/` - NetUtils for working with IP addresses and ports
+* `jkidl/` - Functionality for parsing KIDL spec files
+* `kidl/` - KIDL parser syntax types
+* `kbasejobservice/` - Utils for handling RPC jobs
```

Comment:
I'm pretty sure this is a mock (or something like it) for the NJS. I'm not exactly sure when or how it's used, but believe it's used in the tests.